libraries/chain/controller.cpp

```diff
@@ -3999,15 +3986,95 @@ struct controller_impl {
       bsp->verify_qc(qc_proof);
    }
 
+   // Verify QC block extension and finality block header extension on a Legacy
+   // or Transition block.
+   void verify_legacy_block_exts( const signed_block_ptr& b, const block_header_state_legacy& prev ) {
+      uint32_t block_num = b->block_num();
+
+      auto block_exts = b->validate_and_extract_extensions();
+      auto qc_ext_id = quorum_certificate_extension::extension_id();
+      bool qc_extension_present = block_exts.count(qc_ext_id) != 0;
+      EOS_ASSERT( !qc_extension_present,
+                  block_validate_exception,
```

Comment:
I think all of these in this method should be `invalid_qc_claim`. Also I think it would be more readable to put the `invalid_qc_claim` on the same line as the first arg (styles vary here obviously, but seems that is more common if it fits).